content/blueprints/gcp.md

```diff
@@ -0,0 +1,1277 @@
+---
+layout: bt_wiki
+title: Google Cloud Plugin
+category: Plugins
+draft: false
+weight: 100
+---
+{{% gsSummary %}} {{% /gsSummary %}}
+
+The GCP plugin allows users to use Cloudify to manage cloud resources on GCP. See below for currently supported resource types.
+
+Be aware that some services and resources vary in availability between regions and accounts.
+
+
+# Plugin Requirements
+
+* Python versions:
+  * 2.7.x
+* [GCP](https://cloud.google.com/) account
+
+
+# Compatibility
+
+The GCP plugin uses the official [Google API Python Client](https://github.com/google/google-api-python-client).
+
+
+
+# Terminology
+
+* Region refers to a general geographical area, such as "Central Europe" or "East US".
+* Zone refers to a distinct area within a region. Zones are usually referred to as '{region}-{zone}, i.e. 'us-east1-b' is a zone within the reigon 'us-east1'.
+
+
+# Types
+
+The following are [node type]({{< relref "blueprints/spec-node-types.md" >}}) definitions. Nodes describe resources in your cloud infrastructure. For more information, see [node type]({{< relref "blueprints/spec-node-types.md" >}}).
+
+## Common Properties
+
+All cloud resource nodes have common properties:
+
+**Properties**
+
+Every time you manage a resource with Cloudify,
+it creates one or more connections to the GCP API.
+You specify the configuration for these clients using the `gcp_config` property.
+It should be a dictionary, with the following values:
+
+  * `project` the name of your project on GCP.
+  * `zone` the default zone which will be used,
+    unless overridden by a defined zone/subnetwork.
+  * `auth` the JSON key file provided by GCP.
+    Can either be the contents of the JSON file, or a file path.
+    This should be in the format provided by the GCP credentials JSON export (https://developers.google.com/identity/protocols/OAuth2ServiceAccount#creatinganaccount)
+  * (optional) `network` the default network to place network-scoped nodes in.
+    The default network (`default`) will be used if this is not specified.
+
+Example
+{{< gsHighlight  yaml  >}}
+...
+node_types:
+  my_vm:
+    type: cloudify.gcp.nodes.Instance
+    properties:
+      image_id: <GCP image ID>
+      gcp_config:
+        project: a-gcp-project-123456
+        zone: us-east1-b
+        auth: <GCP auth JSON file>
+{{< /gsHighlight >}}
+
+
+## Using Existing Resources
+
+Many Cloudify GCP types have a property named `use_external_resource`, which defaults to `false`. When set to `true`, the plugin will apply different semantics for each of the operations executed on the relevant node's instances:
+
+  If `use_external_resource` is `true`, then the required properties for that type (`name`, possibly `region` or `zone`) will be used to look up an existing entity in the GCP project.
+  If the entity is found, then its data will be used to popluate the cloudify instance's attributes (`runtime_properties`). If it is not found then the blueprint will fail to deploy.
```

Comment:
If the entity is found, ~~then~~ its data will be used to ~~popluate~~ **populate** the ~~cloudify~~ **Cloudify** instance's attributes (`runtime_properties`). If it is not found **,** ~~then~~ the blueprint will fail to deploy.